beta-src/src/components/ui/WDMain.tsx

```diff
@@ -251,6 +256,116 @@ const WDMain: React.FC = function (): React.ReactElement {
       isLivePhase: false,
     };
   };
+
+  const findUnitInTerritory = (territoryID: string): string | undefined => {
+    const asArray = Object.entries(data.data.units);
+    const findUnit = asArray.find(
+      (elem) =>
+        elem["1"].countryID === String(status.countryID) &&
+        elem["1"].terrID === territoryID &&
+        elem["1"].type === "Army",
+    );
+    return findUnit ? findUnit["1"].id : undefined;
+  };
+
+  const findOrderByUnitID = (unitID: string): string | undefined => {
+    const findOrder = data.data.currentOrders?.find(
+      (currentOrder) => currentOrder.unitID === unitID,
+    );
+    return findOrder ? findOrder.id : undefined;
+  };
+
+  // Automatically enter implied moves for CONVOY
+  useEffect(() => {
+    Object.entries(ordersMeta).forEach(([_, orderMeta]) => {
```

Comment:
Wouldn't it make more sense to do this in the code that receives order updates, rather than as a `useEffect` on a UI element?

https://github.com/adamlerer/webDiplomacy/blob/main/beta-src/src/utils/state/gameApiSlice/reducers/processMapClick.ts#L281